PushNotifications/PushNotifications/PushNotificationsStatic.swift

```diff
@@ -0,0 +1,269 @@
+#if os(iOS)
+import UIKit
+import UserNotifications
+#elseif os(OSX)
+import Cocoa
+import NotificationCenter
+#endif
+import Foundation
+
+@objc public final class PushNotificationStatic: NSObject {
+    
+    private override init() {
+        // prevent other people initialising us
+    }
+    
+    private static var instance: PushNotifications?
+    internal static var tokenProvider = Dictionary<String, TokenProvider>()
+    
+    /**
+     Start PushNotifications service.
+
+     - Parameter instanceId: PushNotifications instance id.
+
+     - Precondition: `instanceId` should not be nil.
+     */
+    /// - Tag: start
+    @objc public static func start(instanceId: String) {
+        if (instance == nil) {
+            instance = PushNotifications(instanceId: instanceId)
+        } else if self.instance!.instanceId != instanceId {
+            print("PushNotifications.shared.start has been called before with a different instance id! before: \(self.instance!.instanceId), now: \(instanceId).")
+            print("If you want to use multiple instances, use `PushNotifications(instanceId: instanceId)` directly. For example:")
+            print("```")
+            print("   let pushNotifications1 = PushNotifications(instanceId: instanceId)")
+            print("   pushNotifications1.start()")
+            print("```")
+        }
+        instance?.start()
+    }
+    
+    /**
+     Register to receive remote notifications via Apple Push Notification service.
+
+     Convenience method is using `.alert`, `.sound`, and `.badge` as default authorization options.
+
+     - SeeAlso:  `registerForRemoteNotifications(options:)`
+     */
+    /// - Tag: register
+    @objc public static func registerForRemoteNotifications() {
+        if let staticInstance = instance {
+            staticInstance.registerForRemoteNotifications()
+        } else {
+            fatalError("PushNotifications.shared.start must have been called first")
```

Comment:
We are making a liberal use of `fatalError` because we don't want incorrect use of the API going out to production and it's not an error you'll get if you use the SDK correctly, so it's not something to handle at runtime. Ideally, our SDK wouldn't allow developers to call functions when they are not supposed to, but unfortunately, our SDK doesn't work this way.

If I was redesigning I'd consider the `.start` returning a new object that has the extra methods. That way you enforce this function can only be called after `start`, at compile time.